pandas/tests/io/parser/test_upcast.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+import pytest
+
+from pandas._libs.parsers import (  # type: ignore[attr-defined]
+    _maybe_upcast,
+    na_values,
+)
+
+from pandas import NA
+import pandas._testing as tm
+from pandas.core.arrays import (
+    BooleanArray,
+    FloatingArray,
+    IntegerArray,
+    StringArray,
+)
+
+
+def test_maybe_upcast(any_real_numpy_dtype):
+    # GH#36712
+    if any_real_numpy_dtype == "float32":
+        pytest.skip()
```

Comment:
Could you include the reason here?

Reply:
Added

Reply:
Sorry, that reason is still not fully clear to me?

Reply:
Ah, you mean it is not in `pandas._libs.parsers.na_values`. We can't add it there instead of skipping here?

Reply:
Should have been more explicit. We are extracting the appropriate na value via ``_compute_na_values()`` that dict does not have a ``float32`` entry.

This is probably a bug, skipping for now but will look into this.

Reply:
Actually this is a bug that was not visible before, fixed it now